test/cli/test_cli_functions.sh

```diff
@@ -284,10 +284,89 @@ function obc_cycle {
     echo_time "✅  obc cycle is done"
 }
 
+function crd_cycle {     
+    echo_time "Starting the crd cycle" 
+    local crd_create_array=($(test_noobaa silence crd create &>/dev/stdout | awk '{print $6}' | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    local crd_status_array=($(test_noobaa silence crd status &>/dev/stdout | awk '{print $6}' | grep -v "Exists" | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    crd_array=($(kubectl get crd | awk '{print $1}' | grep -v "NAME"))
+    
+    #comparing crd status and create arrays
+    echo_time "checking if crd noobaa create command equals to crd status command"
+    for crd_status_array in ${crd_array[@]}
+    do
+        if [[ ${crd_create_array[@]} =~ ${crd_status_array} ]]
+        then            
+            echo_time "✅ ${crd_status_array} exists in the crds list"
+        else
+            echo_time "❌ ${crd_status_array} is not existed in the crds list"         
+        fi               
+    done
+         
+    kuberun get crd  
+    if [ $? -ne 0 ]
+    then
+        echo_time "❌ kubectl get crd failed"
+        exit 1 
+    else 
+        echo_time "✅ kubectl get crd passed"    
+    fi      
```

Comment:
This is checking the kubrun function exit code.
why do we need to check kubrun/kubectl?

Reply:
Asking again to make sure - you told me to run kubectl after every noobaa command i'm checking. Should i just run it as it is or should i check its' exit status? Or is just comparing between noobaa crd and kubectl get crd enough ?

Reply:
We need to check that the kubectl returns all the expected crds.
so, we should have a list of crd (that is only noobaa’s), and then check that all appear in the kubectl get crd.
Cacking the exit code of our function does not do that.